components/eamxx/src/physics/p3/tests/CMakeLists.txt

```diff
@@ -83,12 +83,14 @@ if (NOT SCREAM_P3_SMALL_KERNELS)
 
   # Make sure that a diff in the two implementation triggers a failed test (in debug only)
   # No need to run lots of different thread counts.
-  CreateUnitTest (p3_sk_tests_fail p3_rain_sed_unit_tests.cpp
-    LIBS p3_sk p3_test_infra
-    EXE_ARGS "--flags=\\'${BASELINE_FILE_ARG}\\'"
-    COMPILER_CXX_DEFS SCREAM_FORCE_RUN_DIFF
-    LABELS "p3_sk;physics;fail"
-    ${FORCE_RUN_DIFF_FAILS})
+  if (SCREAM_ENABLE_BASELINE_TESTS)
+    CreateUnitTest (p3_sk_tests_fail p3_rain_sed_unit_tests.cpp
+      LIBS p3_sk p3_test_infra
+      EXE_ARGS "--flags=\\'${BASELINE_FILE_ARG}\\'"
```

Comment:
I am not sure this is the way `--flags` is to be used in ekat's test main. This is meant to be used for stuff like "verbose", and is meant to set a "bool" in the test session. Here, you are trying to pass actual executable flags, but catch2 will likely complain about them, since they are not recognized.

Reply:
For run_and_cmp, this works b/c it uses its own main. Here, we need to work around it.

- For boolean, you can use stuff like `--flags generate,verbose`, which will set `"verbose"` and `"generate"` as valid keys in the map string->bool in the test session
- For args that are not boolean, we need to use `--ekat-test-params` (which I want to sometime shorten to `--params`), and the syntax could be `--ekat-test-params baseline-dir=${SCREAM_BASELINES_DIR}/data`.

Reply:
I tried a simple program that prints the test session flags/params:
```c++
  std::cout << "flags:\n";
  for (auto it : ts.flags) {
    std::cout << it.first << " -> " << it.second << "\n";
  }
  std::cout << "params:\n";
  for (auto it : ts.params) {
    std::cout << it.first << " -> " << it.second << "\n";
  }
  std::cout << "DONE!\n";
```
I got
```bash
$ ./test --flags='-c -b /path/to/something'
Calling initialize_kokkos
 ExecSpace name: Serial
 ExecSpace initialized: yes
 active avx set: -AVX512F-AVX2-AVX
 compiler id: GCC
 FPE support is disabled
 #host threads: 1

Starting catch session on rank 0 out of 1
flags:
-c -b /path/to/something -> 1
params:
DONE!
```

Reply:
The core pb is that catch2 does not let you pass arbitrary flags. So I had to come up with the whole `--flags f1,f2,f3 --ekat-test-params p1=v1,p2=v2,p3=v3` syntax...